File: 2012/submissions/Khalil Fazal/1/Python/Main.py
#! /usr/bin/env python3

import sys

def firstIndex(word, input):
    if len(input) > 0:
        for i in range(0, len(input)):
            j = 0

            while j < len(word) and i + j < len(input) and input[i + j] == word[j]:
                j += 1

            if j == len(word):
                return i

    return -1

def main():
    if len(sys.argv) > 1:
        for input in sys.stdin:
            print(firstIndex(sys.argv[1], input))

if __name__ == "__main__":
    main()

File: 2012/submissions/Khalil Fazal/1/PHP/Main.php
#! /usr/bin/env php
<?
    function firstIndex($word, $input) {
        for ($i = 0; $i < strlen($input); $i++) {
            $j;

            for ($j = 0; $j < strlen($word) and $i + $j < strlen($input) and $input{$i + $j} == $word{$j}; $j++);

            if ($j == strlen($word)) {
                return $i;
            }
        }

        return -1;
    }

    if (count($argv) > 1) {
        $f = fopen('php://stdin', 'r');

        while ($input = fgets($f)) {
            echo firstIndex($argv[1], $input) . PHP_EOL;
        }

        fclose($f);
    }
?>
